deepchem/models/lightning/trainer.py

```diff
@@ -0,0 +1,350 @@
+                Run a fast development run with limited batches for debugging.
+        For all available options, see: https://lightning.ai/docs/pytorch/stable/common/trainer.html#init
+
+
+    Examples
+    --------
+    >>> import deepchem as dc
+    >>> import lightning as L
+    >>> from deepchem.models.lightning.trainer import LightningTorchModel
+    >>> tasks, datasets, _ = dc.molnet.load_clintox()
+    >>> _, valid_dataset, _ = datasets
+    >>> model = dc.models.MultitaskClassifier(
+    ...     n_tasks=len(tasks),
+    ...     n_features=1024,
+    ...     layer_sizes=[1000],
+    ...     dropouts=0.2,
+    ...     learning_rate=0.0001,
+    ...     device="cpu",
+    ...     batch_size=16
+    ... )
+    >>> trainer = LightningTorchModel(
+    ...     model=model,
+    ...     batch_size=16,
+    ...     max_epochs=30,
+    ...     accelerator="cpu",
+    ...     log_every_n_steps=1,
+    ...     fast_dev_run=True
+    ... )
+    >>> # Train with custom checkpoint settings
+    >>> trainer.fit(valid_dataset,
+    ...              max_checkpoints_to_keep=3,
+    ...              checkpoint_interval=1000)
+    >>> predictions = trainer.predict(valid_dataset)
+    >>> trainer.save_checkpoint("model.ckpt")
+    >>> # To reload:
+    >>> trainer2 = LightningTorchModel.load_checkpoint("model.ckpt", model=model)
+    """
+
+    def __init__(self,
+                 model: TorchModel,
+                 batch_size: int = 32,
+                 model_dir: Optional[str] = "default_model_dir",
+                 **trainer_kwargs: Any) -> None:
+
+        self.model: TorchModel = model
+        self.batch_size: int = batch_size
+        self.trainer_kwargs: Dict[str, Any] = trainer_kwargs
+
+        assert model.batch_size == batch_size, \
+            "Model's batch size must match the LightningTorchModel's batch size."
+
+        # checkpointing is enabled by default
+        if 'enable_checkpointing' not in self.trainer_kwargs:
+            self.trainer_kwargs['enable_checkpointing'] = True
+        else:
+            if not self.trainer_kwargs['enable_checkpointing']:
+                model_dir = None  # Disable checkpointing if explicitly set to False
+
+        # Set default_root_dir for Lightning to use our model_dir if not specified
+        if 'default_root_dir' not in self.trainer_kwargs:
+            self.trainer_kwargs['default_root_dir'] = model_dir
+
+        # Create the Lightning module
+        self.lightning_model: DCLightningModule = DCLightningModule(model)
+
+        # Initialize the base Model class with model_dir to ensure compatibility
+        super(LightningTorchModel, self).__init__(model=model,
+                                                  model_dir=model_dir)
+
+    def fit(self,
+            train_dataset: Dataset,
+            max_checkpoints_to_keep: int = 5,
+            checkpoint_interval: int = 1000,
+            num_workers: int = 4,
+            ckpt_path: Optional[str] = None):
```

Comment:
also, in deepchem torch models, fit can be called for 1 specific epoch, hence if there is a restore option within fit to restart fit from latest checkpoint in model_dir. Can't something similar be done here instead? ckpt_path can be given from model_dir if restore = True.